### PR TITLE
HW3_Yun_Young

### PR DESCRIPTION
**Pull request by yyun1**
This is HW3 by Yun, Young
I had most of the problem working, but had trouble doing part f) and g) of the last problem.

### CODE REVIEW

File: 03_good_programming_R_functions/HW3_Yun_Young.Rmd [hunk 1]
```diff
@@ -0,0 +1,250 @@
+---
+title: "HW3_Yun_Young"
+output:
+  html_document: default
+  pdf_document: default
+---
```
Comment (rsettlage) [inline 03_good_programming_R_functions/HW3_Yun_Young.Rmd]: one of the above two should be default, not both, I prefer pdf as the final output, so please remove the html line in future code submission

File: 03_good_programming_R_functions/HW3_Yun_Young.Rmd [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+
+#boxplots all columns in devData_table, which are all dev1 and dev2 by observer
+boxplot(devData_Ob_list_Bind)
+
+## c) A violin plot of all the sd to compare the spread of sd from dev1 to dev2
+
+#violin plot that plots sd_dev1 and sd_dev2 for all 13 observers
+vioplot(table$sd_dev1, table$sd_dev2)
+
+```
```
Comment (rsettlage) [inline 03_good_programming_R_functions/HW3_Yun_Young.Rmd]: Ok, so you have commented your code in detail, that is great.  This is not what you would hand over to a collaborator.  What you would give them is more like:

some informative text about what you are doing in the next code chunk

a code chunk that is commented that describes the couple of steps you are taking

some more text about what just happened and what you are going to do in the next code chunk

etc etc

File: 03_good_programming_R_functions/HW3_Yun_Young.Rmd [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+
+#print out the second table
+print("Data read by Doctor: ")
+data.frame(readingData_Doc)
+
+#print out the summary for table 1 and table 2
+knitr::kable(summary(readingData_Dev), caption="Reading by Device")
+knitr::kable(summary(readingData_Doc), caption="Reading by Doctor")
+
+```
```
Comment (rsettlage) [inline 03_good_programming_R_functions/HW3_Yun_Young.Rmd]: Nice, you thought about the data, not many do.

Out of curiosity, if I had asked you to make a lm to determine if there is a difference between dev and doc, what would have been your next step?

File: 03_good_programming_R_functions/HW3_Yun_Young.Rmd [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+#loop each time using Newton's method to generate next x until the rest of the vector x is filled and plot each point on graph
+for(i in 1:19) 
+{
+  x[i+1] <- x[i] - (3^x[i] - sin(x[i]) + cos(5*x[i]))/(3*x[i]^2 - cos(x[i]) - 5*sin(5*x[i]))
+  points(x[i+1], 3^x[i+1] - sin(x[i+1]) + cos(5*x[i+1]))
+}
+
+#print the estimated values of x for the first 20 estimates
+print("Estimated values of x[1] through x[20] are :")
+print(x)
```
Comment (rsettlage) [inline 03_good_programming_R_functions/HW3_Yun_Young.Rmd]: Where's the function for newton's method?

File: 03_good_programming_R_functions/HW3_Yun_Young.Rmd [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+
+
+## B. Merge the data set by plate and ID
+
+#merge two data frame by "Plate"
+Car_Merged_byPlate <- merge(Car_Person_select, Car_Geconstat_select_2017, by="Plate", all=TRUE)
+
+#merge two data frame by "ID"
+Car_Merged_byID <- merge(Car_Merged_byPlate, Car_Gebreken_select, by="ID", all=TRUE)
+
```
Comment (rsettlage) [inline 03_good_programming_R_functions/HW3_Yun_Young.Rmd]: You probably don't want all=T in both of these cases.  You filtered by year, so in effect removed some license plates from consideration, no need to put them back in the data frame, ditto for the second merge.  Out of curiosity, what computer did you use, how much RAM, and how long did the merging take?

File: 03_good_programming_R_functions/HW3_Yun_Young.Rmd [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+print(c("5 Most frequent defects translated: ", "1. Tire(s) present with a profile depth of 1.6 to 2.5 mm",  "2. Operation / Condition Required Light / Retroreflector 5. *. 55", "3. Excessive oil leakage", "4. Band onvoldoende profiel", "5. Mechanische delen van het remsysteem vertonen slijtage"))
+
+
+#Selects the observations of each of the most frequent defects and store them in different data.
+Car_Defect_1 <- Car_clean[grep(as.factor("aanwezig met een profieldiepte van"), Car_clean$`Inspection Detail`), ]
+Car_Defect_2 <- Car_clean[grep(as.factor("Werking/toestand verplicht licht/retroreflector 5.*.55"), Car_clean$`Inspection Detail`), ]
+Car_Defect_3 <- Car_clean[grep(as.factor("Overmatige olielekkage"), Car_clean$`Inspection Detail`), ]
+Car_Defect_4 <- Car_clean[grep(as.factor("Band onvoldoende profiel"), Car_clean$`Inspection Detail`), ]
+Car_Defect_5 <- Car_clean[grep(as.factor("Mechanische delen van het remsysteem vertonen slijtage"), Car_clean$`Inspection Detail`), ]
```
Comment (rsettlage) [inline 03_good_programming_R_functions/HW3_Yun_Young.Rmd]: This is probably not the best way.  You have the defect code for the 5 most frequent defects, you can use the %in% function to filter for just those.

File: 03_good_programming_R_functions/HW3_Yun_Young.Rmd (hunk 1 above)
Comment (yyun1) [reply 03_good_programming_R_functions/HW3_Yun_Young.Rmd]: I will for next time! But both of my laptop and desktop have problem knitting to PDF. When I knit to pdf, R asks me to install a package, but it does not get installed and I am not able to proceed further.

File: 03_good_programming_R_functions/HW3_Yun_Young.Rmd (hunk 3 above)
Comment (yyun1) [reply 03_good_programming_R_functions/HW3_Yun_Young.Rmd]: I would comapre each reading by dev and doc: lm(Device_1 ~ Day, readingData_Dev) and lm(Doctor_1 ~ Day, readingData_Doctor), then with Device_2 and Doctor_2

File: 03_good_programming_R_functions/HW3_Yun_Young.Rmd (hunk 5 above)
Comment (yyun1) [reply 03_good_programming_R_functions/HW3_Yun_Young.Rmd]: I used Desktop PC, and it did not take much time, compared to reading in the whole data set initially.

**Pull request closed without merging**